src/components/panels/ConfigPanel/TextAreaConfig.tsx

```diff
@@ -26,6 +28,27 @@ export default function TextAreaConfig({
     inputValue = config.value;
   }
 
+  useEffect(() => {
+    if (descriptionRef.current?.firstChild) {
+      setMaxDescriptionHeight(
+        `${
+          (descriptionRef.current.firstChild as HTMLDivElement).clientHeight
+        }px`
+      );
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!descriptionRef.current) {
+      return;
+    }
+    if (showDescription) {
+      descriptionRef.current.style.maxHeight = maxDescriptionHeight;
```

Comment:
This logic is duplicated

Reply:
Yeah, actually a lot of it is duplicated between the two config types. I was going to extract them out but I'm going to wait until after the Formik conversion is completed to avoid duplicating work.